### ISSUE TITLE
Add iClickTheElement function

### ISSUE THREAD

**Issue by emteknetnz** [silverstripe-labs/silverstripe-behat-extension#200]
iClickTheElement is a very useful function that should be in behat-extension

Previously have put a iClickTheElement directly few modules

- https://github.com/silverstripe/silverstripe-asset-admin/blob/1/tests/behat/src/FixtureContext.php#L360
- https://github.com/silverstripe/silverstripe-campaign-admin/blob/1/tests/behat/src/FixtureContext.php#L239
- https://github.com/silverstripe/silverstripe-userforms/blob/5/tests/behat/src/FixtureContext.php#L22
- https://github.com/silverstripe/silverstripe-session-manager/pull/51/files#diff-d9a51caf6b7547eddd66b8c12469c1c423030196fb99bbf11566575dff0370c7R31

However it should live in behat-extension instead

When adding to behat-extension, need to consider any possible conflicts between having the same function on both behat-extension and the module

Wrapping the function in a `if (function_exists($class, 'iClickTheElement')` conditional may help

Need to consider installing new and old version modules alongside new and old versions of behat-extension

### Acceptance Criteria
- [x] The new "iClickTheElement" is added to behat extensions
- [x] The existing "iClickTheElement" is removed from the modules listed above
- [x] Ensure silverstripe/recipe-testing is updated
- [x] The the modules listed above still work nicely once the function is in behat-extensions (it may need additional work here) 
- [ ] Previous minor versions of the modules listed above still work with the new version of behat-extensions (once it's updated to use the new function)

### PRs
- [x] https://github.com/silverstripe/silverstripe-behat-extension/pull/201
- [x] https://github.com/silverstripe/silverstripe-behat-extension/pull/202 (tag 4.4.0 after merge - updated: tag was created)
- [x] https://github.com/silverstripe/recipe-testing/pull/8
- [x] https://github.com/silverstripe/silverstripe-admin/pull/1194
- [x] https://github.com/silverstripe/silverstripe-asset-admin/pull/1191
- [x] https://github.com/silverstripe/silverstripe-campaign-admin/pull/195
- [x] https://github.com/silverstripe/silverstripe-cms/pull/2649
- [x] https://github.com/silverstripe/silverstripe-session-manager/pull/61
- [x] https://github.com/silverstripe/silverstripe-userforms/pull/1057

**Comment by emteknetnz** [comment]
Behat really doesn't like iClickTheElement being in both behat-extension AND a module Feature/FixtureContext - specifically duplicated docblock regex `@Given /^I click the "([^"]+)" element$/` causes the behat to simply not work.  Changing the regex in one location causes behat to function again.  The name of the function doesn't matter because it's in different classes.

Given this, I'm not sure we can add this function into behat-extension as is?  For new versions of modules this is not a problem since we can just remove iClickTheElement from modules because its a dev-only thing and semver doesn't apply.

The issue is that older versions of modules (i.e. 1.6.x-dev) have dev requirements of recipe-testing:^1 or behat-extension:^4.  This means they'll get the latest version of behat-extension that would have `@Given /^I click the "([^"]+)" element$/`, but they'd still have their old version of the `@Given /^I click the "([^"]+)" element$/` which would conflict and cause behat to stop working.  This feature isn't close to important enough to justify backporting.

Releasing this as a new major would sort this, though it seems a little extreme.  We could update recipe-testing to use behat-extension:^5 so most things would auto-update, though any locations that still simply require behat-extenions:^4 would need updating

I don't particularly want to change the regex for the behat-extension version to something else because it reads correctly as is.

We'd need to change this to something like

`@Given /^I click on the "([^"]+)" element$/`
(add the word "on")

Which, reads a bit wrong?  You don't click "on" elements, you just click them

You could argue it's just semantics and it really doesn't matter because it's dev only

Realistically I think we'd need to do this

**Comment by emteknetnz** [comment]
Based on https://english.stackexchange.com/questions/53794/when-to-use-click-and-click-on

'Click on' may actually be actually be more semantically correct

tl;dr

You 'click' a mouse button

You 'click on' a text link using a mouse button
